tests/manifests/test_pod_idempotency.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2024 New Vector Ltd
+#
+# SPDX-License-Identifier: AGPL-3.0-only
+
+import random
+
+import pytest
+import yaml
+
+from . import values_files_to_test
+from .utils import helm_template, template_id
+
+
+@pytest.mark.parametrize("values_file", ["element-web-minimal-values.yaml"])
+@pytest.mark.asyncio_cooperative
+async def test_values_file_renders_idempotent_pods(release_name, values, helm_client, temp_chart):
+    async def _patch_version_chart():
+        with open(f"{temp_chart}/Chart.yaml") as f:
+            chart = yaml.safe_load(f)
+        with open(f"{temp_chart}/Chart.yaml", "w") as f:
+            chart["version"] = f"{random.randint(0, 100)}.{random.randint(0, 100)}.0"
+            yaml.dump(chart, f)
+        return await helm_client.get_chart(temp_chart)
+
+    first_render = {}
+    second_render = {}
+    for template in await helm_template(
+        (await _patch_version_chart()), release_name, values, has_service_monitor_crd=True, skip_cache=True
+    ):
+        if template["kind"] in ["Deployment", "StatefulSet", "Job"]:
```

Comment:
I think we should be testing everything. We should test that all manifests aren't equal by default and that if we remove `metadata.labels['helm.sh/chart']` all manifests are now equal. That proves that the label isn't on the Pod spec and it also proves that all hashes (be it on the Pod spec or the parent) are avoiding including the full manifest